tests/test_md.py

```diff
@@ -1011,3 +1011,96 @@ def test_local_md_nonbonded_all_pairs_subset():
 
     np.testing.assert_array_equal(ref_local_xs, test_local_xs)
     np.testing.assert_array_equal(ref_local_boxes, test_local_boxes)
+
+
+def test_context_invalid_boxes():
+    """Verify that nonbonded all pairs potentials provided to the context will correctly validate the box size"""
+    mol, _ = get_biphenyl()
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+
+    temperature = constants.DEFAULT_TEMP
+    dt = 1.5e-3
+    friction = 0.0
+    seed = 2022
+    steps = 1
+    rng = np.random.default_rng(seed)
+
+    unbound_potentials, sys_params, masses, coords, box = get_solvent_phase_system(mol, ff, 0.0, minimize_energy=False)
+    v0 = np.zeros_like(coords)
+
+    ligand_idxs = np.arange(len(coords) - mol.GetNumAtoms(), len(coords), dtype=np.int32)
+    reference_idx = rng.choice(ligand_idxs)
+    selection = np.array(list(set(ligand_idxs).difference(set([reference_idx]))), dtype=np.int32)
+
+    bps = []
+    for p, bp in zip(sys_params, unbound_potentials):
+        bound_impl = bp.bind(p).to_gpu(np.float32).bound_impl
+        bps.append(bound_impl)
+
+    intg = LangevinIntegrator(temperature, dt, friction, masses, seed)
+
+    ctxt = custom_ops.Context(coords, v0, box, intg.impl(), bps)
+    ctxt.multiple_steps(steps)
+    ctxt.multiple_steps_U(steps, 0, 0)
+    ctxt.multiple_steps_local(steps, ligand_idxs, burn_in=0)
+    ctxt.multiple_steps_local_selection(steps, reference_idx, selection, burn_in=0)
+
+    # Make the box way too small, which should trigger the failure
+    ctxt.set_box(box * 0.01)
+    err_msg = "cutoff with padding is more than half of the box width, neighborlist is no longer reliable"
+    with pytest.raises(RuntimeError, match=err_msg):
+        _, boxes = ctxt.multiple_steps(steps)
+        assert len(boxes) == 1
+    with pytest.raises(RuntimeError, match=err_msg):
+        _, boxes, _ = ctxt.multiple_steps_U(steps, 0, 0)
+        assert len(boxes) == 1
+    with pytest.raises(RuntimeError, match=err_msg):
+        _, boxes = ctxt.multiple_steps_local(steps, ligand_idxs, burn_in=0)
+        assert len(boxes) == 1
+    with pytest.raises(RuntimeError, match=err_msg):
+        _, boxes = ctxt.multiple_steps_local_selection(steps, reference_idx, selection, burn_in=0)
+        assert len(boxes) == 1
+
+    # Without returning boxes no check will be performed
```

Comment:
I'm not sure I understand why this behavior is desirable? Wouldn't we want to perform this check in all cases?

Reply:
Was trying to avoid having to pay the cost of copying if the user doesn't request the boxes. Probably doesn't make much difference, would probably only matter where lots of `multiple_steps` are called with few steps and no coords/boxes are returned.

Reply:
OK, I think this makes sense. The only situation I can think of where we might run a large number of steps without storing any frames is during equilibration/burn-in, but maybe this is not an issue in practice. Should we make a note in the docstring for `multiple_steps_*` that checks are only performed when storing frames?

Reply:
Will update docstrings

Reply:
Added in https://github.com/proteneer/timemachine/pull/1053/commits/a77d8249e0107b4773cc35d50ef5b25aeb72ff4c

Reply:
Is it worth also adding an explicit warning like "note: runtime validation of box size is only performed when collecting frames, and thus will not be performed (for example) when `store_x_interval > n_steps`.", maybe to the docstring for `store_x_interval`?

Reply:
Expanded in https://github.com/proteneer/timemachine/pull/1053/commits/5b1fe2ae7866b4ea182958baef753fc7b764e81a